dbt-adapters/src/dbt/adapters/base/impl.py

```diff
@@ -872,6 +872,7 @@ def valid_snapshot_target(
         # aren't always present.
         for column in ("dbt_scd_id", "dbt_valid_from", "dbt_valid_to"):
             desired = column_names[column] if column_names else column
+            desired = desired.lower() if desired is not None else desired
```

Comment:
Do we specifically want to change the case of the column name that we're appending? Or do we just want to do a case-insensitive search to know whether to append it? For example:
```python
for column in ("dbt_scd_id", "dbt_valid_from", "dbt_valid_to"):
    desired = column_names[column] if column_names else column
    if desired and desired.lower() not in names:
        missing.append(desired)
```

Reply:
Agreed. Done!